Documentation/rbd-mirroring.md

```diff
@@ -0,0 +1,247 @@
+---
+title: RBD Mirroring
+weight: 3242
+indent: true
+---
+
+# RBD Mirroring
+
+[RBD mirroring](https://docs.ceph.com/en/latest/rbd/rbd-mirroring/)
+ is an asynchronous replication of RBD images between multiple Ceph clusters.
+ This capability is available in two modes:
+
+* Journal-based: Every write to the RBD image is first recorded
+ to the associated journal before modifying the actual image.
+ The remote cluster will read from this associated journal and
+ replay the updates to its local image.
+* Snapshot-based: This mode uses periodically scheduled or
+ manually created RBD image mirror-snapshots to replicate
+ crash-consistent RBD images between clusters.
+
+## Table of Contents <!-- omit in toc -->
+
+* [Create RBD Pools](#create-rbd-pools)
+* [Bootstrap Peers](#bootstrap-peers)
+* [Configure the RBDMirror Daemon](#configure-the-rbdmirror-daemon)
+* [Enable CSI Replication Sidecars](#enable-csi-replication-sidecars)
+* [Volume Replication Custom Resources](#volume-replication-custom-resources)
+
+## Create RBD Pools
+
+In this section we create specific rbd pools that are RBD mirroring
+ enabled for use with the DR use case.
+
+> :memo: **Note:** It is also feasible to edit existing pools and
+> enable them for replication.
+
+Execute the following steps on each peer cluster to create mirror
+ enabled pools:
+
+* Create a RBD pool that is enabled for mirroring by adding the section
+ spec.mirroring in the CephBlockPool CR:
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: mirroredpool
+  namespace: rook-ceph
+spec:
+  replicated:
+    size: 1
+  mirroring:
+    enabled: true
+    mode: image
+    # schedule(s) of snapshot
+    snapshotSchedules:
+      - interval: 24h # daily snapshots
+        startTime: 14:00:00-05:00
+```
+
+```bash
+kubectl create -f pool-mirrored.yaml
+```
+
+> ```bash
+> cephblockpool.ceph.rook.io/mirroredpool created
+> ```
+
+* Repeat the steps on the peer cluster.
+
+> :memo: WARNING: Pool name across the cluster peers should be the same
+> for RBD replication to function.
+
+For more information on CephBlockPool CR, please refer to the [ceph-pool-crd documentation](Documentation/ceph-pool-crd.md#mirroring).
+
+## Bootstrap Peers
+
+In order for the rbd-mirror daemon to discover its peer cluster, the
+ peer must be registered and a user account must be created.
+
+The following steps enable bootstrapping peers to discover and
+ authenticate to each other:
+
+* For Bootstrapping a peer cluster its bootstrap secret is required. To determine the name of the secret that contains the bootstrap secret execute the following command on the remote cluster (site-b)
```

Comment:
```suggestion
* For Bootstrapping a peer cluster its bootstrap secret is required. To determine the name of the secret that contains the bootstrap secret execute the following command on the remote cluster (cluster-b)
```
since primary cluster is named `cluster-a`